### DISCUSSION TITLE
Converting batch to Linux / OSX. Help required.

### DISCUSSION THREAD

**Discussion by Lamer87** [Lamer87/Space_Engineers_DLC_unlocker#45] [Q&A]
Linux/OSX version removed because:
1- it was not doing any checks
2- it only worked for installing, regardless of whether files were correct
3- it's not up to date with the current cmd version

I don't have the knowledge to convert it, but if someone has it and wants to help,
here you can find a clean 1.1 version of the cmd, without the color codes:

[clean cmd v1.1.txt](https://drive.google.com/file/d/1htmFTLW4PMsnbTut9Haf6wBqpFktsx1D/view?usp=sharing)

And here you can find my try to fix the linux one (from the linux version 1.0, no "update check" and "update zone"):

[linux v1.0.txt](https://drive.google.com/file/d/1S8iTSwmNwNwQ-QKlIbFh8p-nLRyyWqwz/view?usp=sharing)

**Comment by MisterFreebie** [comment]
I dont see anythink like codes or texts

**Reply by Lamer87** [reply]
lol thanks for the answer, can you upload a screenshot please? I really want to see what is up with this version

**Comment by TannerCS** [comment]
> lol thanks for the answer, can you upload a screenshot please? I really want to see what is up with this version

I can't seem to get it working on Linux. Is this what you're referring to? I'm on Arch

![image](https://github.com/Lamer87/Space_Engineers_DLC_unlocker/assets/22056617/4b658a33-5ddd-4196-9ec8-e006aed95d98)

**Reply by TannerCS** [reply]
File has carriage return characters which don't work on Linux. That's why I couldn't get it working. But now that it is, I can confirm what you're asking

![image](https://github.com/Lamer87/Space_Engineers_DLC_unlocker/assets/22056617/93ca755b-94eb-4e7c-9514-527aa513dbf4)

**Reply by Lamer87** [reply]
Thank you so much for your reply, now I know for sure that color codes mess everything up.
Soon I will release a new version because CreamAPI has been updated, in particular the latest version also affects Space Engineers.
I'll take this opportunity to load a "clean" version of the Linux one, to remove all the superfluous codes, even if I don't have the knowledge to update it with an additional section that I added to the cmd for windows, which detects if the old version of CreamAPI is present and asks if you want to update it.
And, I don't have the knowledge to fix it if it's not working on linux, never used it, it's just that someone took it and converted to linux/osx and I uploaded it without being able to test.
Btw thanks again :)

**Comment by Lamer87** [comment]
Linux/OSX version removed because:
1- it was not doing any checks
2- it only worked for installing, regardless of whether files were correct
3- it's not up to date with the current cmd version

I don't have the knowledge to convert it, but if someone has it and wants to help,
here you can find a clean 1.1 version of the cmd, without the color codes:

[clean cmd v1.1.txt](https://drive.google.com/file/d/1htmFTLW4PMsnbTut9Haf6wBqpFktsx1D/view?usp=sharing)

And here you can find my try to fix it (from the linux version 1.0, no update check and zone):

[linux v1.0.txt](https://drive.google.com/file/d/1S8iTSwmNwNwQ-QKlIbFh8p-nLRyyWqwz/view?usp=sharing)

**Reply by TannerCS** [reply]
Documents can't be accessed. Need to set your file permissions in Google drive. I'll try to take a look and see what I can do. Thanks 🙂

**Reply by Lamer87** [reply]
Sorry, now links must work :)
my biggest problem is that I rely heavily on the "goto" command, which has no linux counterpart.
